node-wasm/src/node.rs

```diff
@@ -38,141 +37,219 @@ pub struct WasmNodeConfig {
     pub bootnodes: Vec<String>,
 }
 
-#[wasm_bindgen(js_class = Node)]
-impl WasmNode {
-    /// Create a new Lumina node.
+#[wasm_bindgen(js_name = NodeClient)]
+struct NodeDriver {
+    _worker: SharedWorker,
+    _onerror_callback: Closure<dyn Fn(MessageEvent)>,
+    channel: WorkerClient,
+}
+
+#[wasm_bindgen(js_class = NodeClient)]
+impl NodeDriver {
+    /// Create a new connection to a Lumina node running in a Shared Worker.
+    /// Note that single Shared Worker can be accessed from multiple tabs, so Lumina may
+    /// already have been started. Otherwise it needs to be started with [`NodeDriver::start`].
     #[wasm_bindgen(constructor)]
-    pub async fn new(config: WasmNodeConfig) -> Result<WasmNode> {
-        let config = config.into_node_config().await?;
+    pub async fn new() -> Result<NodeDriver> {
```

Comment:
since `SharedWorkers` aren't supported on chrome android, can we have `new_shared()` and `new_dedicated()` and somehow switch between the two?
https://issues.chromium.org/issues/40290702